nixos/modules/services/web-apps/rundeck.nix

```diff
@@ -0,0 +1,550 @@
+      "${cfg.dataDir}" = {
+        d = {
+          mode = "0750";
+          user = cfg.user;
+          group = cfg.group;
+        };
+      };
+
+      "${cfg.dataDir}/etc" = {
+        d = {
+          mode = "0750";
+          user = cfg.user;
+          group = cfg.group;
+        };
+      };
+
+      "${cfg.dataDir}/data" = {
+        d = {
+          mode = "0750";
+          user = cfg.user;
+          group = cfg.group;
+        };
+      };
+
+      "${cfg.dataDir}/projects" = {
+        d = {
+          mode = "0750";
+          user = cfg.user;
+          group = cfg.group;
+        };
+      };
+
+      "${cfg.dataDir}/libext" = {
+        d = {
+          mode = "0750";
+          user = cfg.user;
+          group = cfg.group;
+        };
+      };
+
+      "${cfg.dataDir}/var" = {
+        d = {
+          mode = "0750";
+          user = cfg.user;
+          group = cfg.group;
+        };
+      };
+
+      "${cfg.dataDir}/var/logs" = {
+        d = {
+          mode = "0750";
+          user = cfg.user;
+          group = cfg.group;
+        };
+      };
+
+      "${cfg.dataDir}/var/tmp" = {
+        d = {
+          mode = "0750";
+          user = cfg.user;
+          group = cfg.group;
+        };
+      };
+
+      "${cfg.dataDir}/.ssh" = {
+        d = {
+          mode = "0700";
+          user = cfg.user;
+          group = cfg.group;
+        };
+      };
+
+      "${cfg.configDir}" = {
+        d = {
+          mode = "0750";
+          user = cfg.user;
+          group = cfg.group;
+        };
+      };
+
+      "${cfg.configDir}/ssl" = {
+        d = {
+          mode = "0750";
+          user = cfg.user;
+          group = cfg.group;
+        };
+      };
+
+      "${cfg.tempDir}" = {
+        d = {
+          mode = "0750";
+          user = cfg.user;
+          group = cfg.group;
+        };
+      };
+    };
+
+    systemd.services.rundeck = {
+      description = "Rundeck Service";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      wants = lib.optional (cfg.database.type == "mysql") "mysql.service";
+
+      environment = {
+        RDECK_BASE = cfg.dataDir;
+        RUNDECK_CONFIG_DIR = cfg.configDir;
+        JAVA_OPTS = lib.concatStringsSep " " cfg.javaOpts;
+      };
+
+      serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
+        ExecStart = toString (
+          pkgs.writeShellScript "start-rundeck" ''
+            # Generate UUID
+            UUID_FILE="${cfg.dataDir}/.uuid"
+            if [ -z "${cfg.serverUUID}" ] && [ ! -f "$UUID_FILE" ]; then
+              ${pkgs.libuuid}/bin/uuidgen > "$UUID_FILE"
+              chmod 640 "$UUID_FILE"
+              chown ${cfg.user}:${cfg.group} "$UUID_FILE"
+            fi
+
+            # Generate SSH
+            if [ ! -f ${cfg.dataDir}/.ssh/id_rsa ]; then
+              ${pkgs.openssh}/bin/ssh-keygen -t rsa -b 4096 -N "" -f ${cfg.dataDir}/.ssh/id_rsa
```

Comment:
Use ed25519 keys

Reply:
I can't for compatibility reasons.
Rundeck also still uses RSA.

Reply:
Fixed